plotly/tools.py

```diff
@@ -1477,339 +1767,155 @@ def create_scatterplotmatrix(df, useindex = False, index = " ",
-                    for name in trace_list[j]:
-                        fig.append_trace(trace_list[j][name], y_index, x_index)
-                    j += 1
+        Example 3: Adding some style
+        ```
+        from plotly.graph_objs import graph_objs
+        import plotly.plotly as py
+        from plotly.tools import FigureFactory as FF
+
+        import numpy as np
+        import pandas as pd
+
+        # Create 4X4 dataframe with index
+        df = pd.DataFrame(np.random.randn(10, 4),
+                           columns=["A", "B", "C", "D"])
         
-            # Check if length of col_names is equal to the
-            # number of lists in matrix
-            if len(col_names) != dim:
-                raise exceptions.PlotlyError("Your list of variable_"
-                                             "names must match the "
-                                             "number of lists in your "
-                                             "array. That is to say that "
-                                             "both lists must have the "
-                                             "same dimension.")
-
-            # Insert col_names into the figure
-            for j in range(dim):
-                xaxis_place = "xaxis" + str(dim*dim - dim + 1 + j)
-                fig['layout'][xaxis_place].update(title = col_names[j])
-
-            for j in range(dim):
-                yaxis_place = "yaxis" + str(1 + dim*j)
-                fig['layout'][yaxis_place].update(title = col_names[j])
-
-            # Set height and width if not already selected by user
-            if (height == " "):
-                height = 400 + 200*(dim - 1)
-
-            if (width == " "):
-                width = 400 + 200*(dim - 1)
-
-            if diagonal == "Histogram":
-                fig['layout'].update(
-                    height = height, width = width,
-                    title = title,
-                    showlegend = True,
-                    barmode = "stack")
-                return fig
-
-            if diagonal == "Scatter":
-                fig['layout'].update(
-                    height = height, width = width,
-                    title = title,
-                    showlegend = True)
-                return fig
+        df["Fruit"] = pd.Series(["apple","apple","pear","apple","apple",
+                                 "pear","pear","pear","apple","pear"])
+
+        # Create scatterplot matrix
+        fig = FF.create_scatterplotmatrix(df, useindex = True, index = "Fruit",
+                                          symbol = 2, size = 8,
+                                          diag = "Histogram")
+
+        # Plot
+        fig = py.iplot(fig, filename='ScatterPlot Matrix')
+        ```
+        """
+        # TODO: protected until #282
+        matrix = []
+        headers = []
+        index_vals = []
+        global diag_choices
+        diag_choices = ["Scatter", "Histogram", "box"]
```

Comment:
Also, try not to use `global`, just define this at the top of the module:

`DIAG_CHOICES = ('scatter', 'histogram', 'box')`